Do not stop checking DataView properties on first failure

MarkCDavid opened the pull request:
A small fix for the following situation:

```
[what_is_this:: not_a_link]
[what_is_that:: [[that_is_a_link]]]
```

When iterating through the DataView properties, previously, the code would see that the metadata property `what_is_this` is not a link and would stop immediately. If we allow for it to continue, instead of stopping, it will check `what_is_that` property, return it as intended and render the link correctly.

The fix here allows for consideration across all DataView properties, as such, we can have DataView properties with a link and ones without a link together.

MarkCDavid committed:
Do not stop checking DataView properties on first failure

dobien commented:
Thanks! This works.
natefrisch01 Please fix this problem. The plugin is very cool, thanks for it, but this problem really hinders the ability to use it.

dobien commented:
Thanks, this helped

dobien commented:
I simultaneously encountered an issue where some edges weren't processed because the Dataview cache linking note names to paths (or whatever it caches) was outdated after moving files. This likely happened due to `Automatically update internal links` being disabled (possibly) during some period. I used `Dataview: Drop all cached file metadata` and restarted Obsidian to resolve it. Sorry for the offtopic, but I don't know where to write about this. It wasn't too trivial.

cr7pt0gr4ph7 commented on src/linkManager.ts:
```diff
@@ -459,10 +459,11 @@ export class LinkManager {
-                    //metadata is not a link, return null
-                    return null;
+                  // We will continue to check other DataView properties
+                  continue
```
Just some small nitpick :)

```suggestion
                  break;
```

MarkCDavid replied:
Oh, we break out from the switch context, that's correct.

cr7pt0gr4ph7 commented:
@MarkCDavid Thanks a lot! I also stumbled upon this after freshly installing Graph Links Types and scratching my head on why it didn't work...

@natefrisch01 Thank you for the great plugin! Would you be able to merge this fix so that new users do not hit this roadblock in the future? Thanks!

MarkCDavid committed:
Update src/linkManager.ts

Instead of continuing, break out of the switch context and allow the loop to continue "naturally".

Co-authored-by: Lukas Waslowski <cr7pt0gr4ph7@gmail.com>

DataStrategist commented:
I too just hit this problem. If I go to a note where things aren't working... if I delete the headers and a lot of the text, it starts working and adding everything back leaves everything operational. Does that help? 

Also... @MarkCDavid , it seems you may have found a solution that was approved by @cr7pt0gr4ph7 ? Can I install  aversion from your branch?

cx-diao commented:
I have the same issue and it's quite frustrating after I spent an afternoon figuring out why it just doesn't work.

CHerSun commented:
I've tried to include this change on my fork here https://github.com/CHerSun/Graph-Link-Types . ~~Can't seem to get this plugin to work at all, whatever I do - I can't get link descriptions shown (both on local graph and global one)~~ my bad, forgot to reenable it after new installation. Works as expected.

In the fork I've tried to do the upkeeping work - bumped package versions, added automatic pipeline to build release and bump plugin versions, but my knowledge in JS/TS is definitely lacking to actualize the code itself. I'd appreciate the help to revive the plugin and maybe extend functionality.

Plugin from my fork can be added via BRAT using the github link.

natefrisch01 committed:
Merge branch 'master' into fix/consider_all_links_within_a_page

natefrisch01 committed:
Update linkManager.ts

FIxing indenting
